test/utils/TestUtils.ts

```diff
@@ -167,52 +198,51 @@ const deployRelayHub = async (
   );
 };
 
-const deployVerifiers = async (
-  smartWalletFactory: SmartWalletFactory | CustomSmartWalletFactory,
-  isCustom = false
-) => {
-  let deployVerifier: DeployVerifier | CustomSmartWalletDeployVerifier;
-
-  if (isCustom) {
-    const deployVerifierFactory = await ethers.getContractFactory(
-      'CustomSmartWalletDeployVerifier'
-    );
-    deployVerifier = await deployVerifierFactory.deploy(
-      smartWalletFactory.address
-    );
-  } else {
-    const deployVerifierFactory = await ethers.getContractFactory(
-      'DeployVerifier'
-    );
-    deployVerifier = await deployVerifierFactory.deploy(
-      smartWalletFactory.address
-    );
-  }
-
-  const relayVerifierFactory = await ethers.getContractFactory('RelayVerifier');
-  const relayVerifier = await relayVerifierFactory.deploy(
+const deployVerifiers = async <
+  C1 extends SupportedDeployVerifier,
+  C2 extends RelayVerifier | BoltzRelayVerifier | MinimalBoltzRelayVerifier
+>(
+  smartWalletFactory: SupportedSmartWalletFactory,
+  type: SupportedType = 'Default'
+): Promise<{
+  deployVerifier: C1;
+  relayVerifier: C2;
+}> => {
+  const deployVerifierFactory = await ethers.getContractFactory(
+    `${type === 'Default' ? '' : type}DeployVerifier`
+  );
+  const deployVerifier = (await deployVerifierFactory.deploy(
     smartWalletFactory.address
+  )) as C1;
+
+  const relayVerifierFactory = await ethers.getContractFactory(
+    `${type === 'Default' || type === 'Custom' ? '' : type}RelayVerifier`
   );
+  const relayVerifier = (await relayVerifierFactory.deploy(
+    smartWalletFactory.address
+  )) as C2;
 
   return {
     deployVerifier,
     relayVerifier,
   };
 };
 
-const createSmartWalletFactory = async (
-  template: IForwarder,
-  isCustom = false,
+const createSmartWalletFactory = async <T extends SupportedSmartWalletFactory>(
+  template: SupportedSmartWallet | MinimalBoltzSmartWallet,
+  type: SupportedType = 'Default',
```

Comment:
If we declare `type` as the latest argument, we could avoid specifying it when it's value is `Default`, otherwise it doesn't make sense to use a default value here